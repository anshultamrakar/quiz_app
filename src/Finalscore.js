import React from 'react'
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { PieChart } from 'react-minimal-pie-chart';
import {Grid , Paper} from '@material-ui/core';
import { useState, useEffect } from 'react';

import react from './Data/react.json'
import javascript from './Data/javascript.json'
import {useLocation} from 'react-router-dom'

const Finalscore = () => {
const location = useLocation()
const [language]  = useState(location.state.language)

const answer  = location.state.selectedAnswer 

const [rigthAnswer , setRigthAnswer] = useState(0)
const [wrongAnswer , setWrongAnswer ] = useState(0)



useEffect(() => {
  let data
  if(language === "ReactJS"){
     data = react
  }else{
     data = javascript
  }

  const correctAnswer = []
  let totalCorrect = 0
  let totalInCorrect = 0

  data.map((item)=>{
      correctAnswer.push(item.answer)
  })
  
  for(let i=0;i<answer.length;i++){
      if(answer[i] === correctAnswer[i]){
          totalCorrect = totalCorrect+1
      }else if(answer[i] !== correctAnswer[i]){
          totalInCorrect = totalInCorrect+1
      }
  }
  setRigthAnswer(totalCorrect)
  setWrongAnswer (totalInCorrect)
}
,[answer, language])


  return (
    <Paper >
     <header>
        <AppBar position="static" >
             <Toolbar>
             <Typography> Your Score </Typography>
             </Toolbar>
        </AppBar> 
   </header>
      <Grid style = {{ display :"flex", flexDirection : 'column', justifyContent : "center", alignItems : "center" }} >
         <h2>Correct Answers : {rigthAnswer} </h2>
         <br/>
         <h2> Wrong Answers : {wrongAnswer} </h2>
      </Grid>
      <br/>
      <Grid  style = {{marginLeft : "500px"}} item className = "grid-wrapper">
      <PieChart
 

   center={[50, 50]}
   data={[
     {
     color: "#C13C37",
     title: "Correct Answer",
     value:  rigthAnswer ,
     },
     {
     color: "#6A2135",
     title: "Wrong Answer",
     value: wrongAnswer,
     },
   ]}
   labelPosition={72}
   lengthAngle={360}
   lineWidth={15}
   paddingAngle={0}
   radius={50}
 
   startAngle={0}
   viewBoxSize={[300, 300]}
   label={(data) => Math.ceil(data.dataEntry.percentage) + '%'}    
     labelStyle={{
     fontSize: "5px",
     fill: "#3a3a3a",
     fontWeight: "800",
     fontFamily: 'sans-serif',
            }}/>
      </Grid>
    </Paper>
  )
}

export default Finalscore