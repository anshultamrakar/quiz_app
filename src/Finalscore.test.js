/**
 * @jest-environment jsdom
*/


import React from 'react'
import { configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Finalscore from './Finalscore'

configure({ adapter: new Adapter() });


jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/finalscore',
      search: '',
      hash: '',
      state: {
        language : 'ReactJS',
        selectedAnswers : ['Method','Both 1 & 2','Class','Props','Extends']
      },
      key: '5nvxpbdafa',
    }),
  }));



describe('Rendering the finalscore component', () => {
    let wrapper
    it('Renders correctly and set states', () => {
      wrapper = shallow(<Finalscore/>); 
      expect(wrapper.exists()).toBeTruthy();
    });
  
  });