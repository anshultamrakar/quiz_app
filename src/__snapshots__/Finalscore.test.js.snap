// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Finalscore/> cases Renders correctly and set states 1`] = `ShallowWrapper {}`;
